File: 1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.js
/**
 * @param {number[]} prices
 * @return {number[]}
 */
var finalPrices = function(prices) {
  return prices.map((n, i) => {
    if (i === prices.length - 1) return n
    const discount = prices.slice(i+1).find(el => el <= n)
    return discount ? n - discount : n
  })
};

File: 1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.go
func finalPrices(prices []int) []int {
  for i := 0; i < len(prices); i ++ {
    p := prices[i]
    for j := i + 1; j < len(prices); j ++ {
      if prices[j] <= p {
        prices[i] -= prices[j]
        break
      }
    }
  }
  return prices  
}